src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -453,6 +453,66 @@ public function isAbsolutePath($file)
     }
 
     /**
+     * Creates a temporary file with support for custom stream wrappers.
+     *
+     * @param string $dir    The directory where the temporary filename will be created.
+     * @param string $prefix The prefix of the generated temporary filename.
+     *                       Note: Windows uses only the first three characters of prefix.
+     *
+     * @return string The new temporary filename (with path), or false on failure.
+     */
+    public function tempnam($dir, $prefix)
+    {
+        $limit = 10;
+        $schemeSeparator = '://';
+        list($scheme, $hierarchy) = $this->getSchemeAndHierarchy($dir);
+
+        // If no scheme or scheme is "file" create temp file in local filesystem
+        if (null === $scheme || 'file' === $scheme) {
+            $tmpFile = tempnam($hierarchy, $prefix);
+        }
```

Comment:
we can simplify the below [if statements] (https://github.com/markchalloner/symfony/blob/ticket_10018/src/Symfony/Component/Filesystem/Filesystem.php#L475-L484) and remove $schemeSeparator too.
```php
// If no scheme or scheme is "file" create temp file in local filesystem
if (null === $scheme || 'file' === $scheme) {
    $tmpFile = tempnam($hierarchy, $prefix);

    return null === $scheme ? $tmpFile:$scheme.'://'.$tmpFile;
}
```